hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/HazelcastWriters.java

```diff
@@ -57,7 +71,97 @@ private HazelcastWriters() {
 
     @Nonnull
     @SuppressWarnings("unchecked")
-    public static ProcessorMetaSupplier writeMapP(String name, ClientConfig clientConfig) {
+    public static <T, K, V> ProcessorMetaSupplier mergeMapP(
+            @Nonnull String name,
+            @Nullable ClientConfig clientConfig,
+            @Nonnull DistributedFunction<T, K> toKeyFn,
+            @Nonnull DistributedFunction<T, V> toValueFn,
+            @Nonnull DistributedBinaryOperator<V> mergeFn
+    ) {
+        return updateMapP(name, clientConfig, toKeyFn, (V oldValue, T item) -> {
+            V newValue = toValueFn.apply(item);
+            if (oldValue == null) {
+                return newValue;
+            }
+            return mergeFn.apply(oldValue, newValue);
+        });
+    }
+
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    public static <T, K, V> ProcessorMetaSupplier updateMapP(
+            @Nonnull String name,
+            @Nullable ClientConfig clientConfig,
+            @Nonnull DistributedFunction<T, K> toKeyFn,
+            @Nonnull DistributedBiFunction<V, T, V> updateFn
+    ) {
+        boolean isLocal = clientConfig == null;
+        return dontParallelize(new HazelcastWriterSupplier<>(
+                serializableConfig(clientConfig),
+                index -> new ArrayList<>(),
+                ArrayList::add,
+                instance -> buffer ->
+                {
+                    Map<K, T> tmpMap = new HashMap<>();
+                    IMap<K, V> map = instance.getMap(name);
+                    AbstractEntryProcessor<K, V> entryProcessor = new AbstractEntryProcessor<K, V>() {
+                        @Override
+                        public Object process(Entry<K, V> entry) {
+                            V oldValue = entry.getValue();
+                            T item = tmpMap.get(entry.getKey());
+                            V newValue = updateFn.apply(oldValue, item);
+                            entry.setValue(newValue);
+                            return null;
+                        }
+                    };
```

Comment:
You've moved this part to the inner lambda. Now, `tmpMap` and `entryProcessor` will be created every time the buffer is flushed. Before, they were recycled.